src/wp-includes/functions.php

```diff
@@ -4695,6 +4695,23 @@ function _mce_set_direction( $mce_init ) {
 	return $mce_init;
 }
 
+/**
+ * Determines whether the current request is a WordPress REST API request.
+ *
+ * The function relies on the 'REST_REQUEST' global. As such, it only returns true when an actual REST _request_ is
+ * being made. It does not return true when a REST endpoint is hit as part of another request, e.g. for preloading a
+ * REST response. See {@see wp_is_rest_endpoint()} for that purpose.
+ *
+ * This function should not be called until the {@see 'parse_request'} action, as the constant is only defined then,
+ * even for an actual REST request.
+ *
+ * @since 6.5.0
+ *
+ * @return bool True if it's a WordPress REST API request, false otherwise.
+ */
+function wp_is_rest_request(): bool {
```

Comment:
Setting a return type slows this function down by about 20%

Reply:
@sybrew I'm not sure where you get this from. In any case, I am sure the time this function takes is irrelevant in the grand scheme of things, since it's just a constant check.

Reply:
@felixarntz every time a value is returned, its type is checked. This requires more processing time. Simply allowing to pass the value unchecked executes no additional code, so that's faster.

I'm trying to prevent this from becoming the norm. PHP isn't typed and nothing has been done to optimize type hinting thus far for userland. It's only a waste of resources.

Reply:
@sybrew That's fair. It probably requires a larger discussion, not in a specific ticket like this. I'm not sure what I think about it yet, but it looks like this would be the first return type in core anyway, so I'll just leave it out. Done in https://github.com/WordPress/wordpress-develop/pull/5658/commits/32a27c6e103a8a40f576557e5e8874e768de20dc

Reply:
There's also no need to crash a user's site because a wrong type is used during development. Just make sure this doesn't happen; it's easily achievable. We have PHPDoc, static code checks, and unit testing to help us with this. We write the code, so we can ascertain the return type without type hinting.